packages/next/lib/eslint/runLintCheck.ts

```diff
@@ -321,13 +321,17 @@ export async function runLintCheck(
         outputFile
       )
     } else {
-      // Display warning if no ESLint configuration is present during "next build"
+      // Display warning if no ESLint configuration is present inside
+      // config file during "next build", no warning is shown when
+      // no eslintrc file is present
       if (lintDuringBuild) {
-        Log.warn(
-          `No ESLint configuration detected. Run ${chalk.bold.cyan(
-            'next lint'
-          )} to begin setup`
-        )
+        if (eslintrcFile) {
```

Comment:
If `eslintrcFile` exists, wouldn't line 309 be true? 🤔

Reply:
It can have an empty config which is what I think this warning was initially meant for

Reply:
I think the bug is here: https://github.com/vercel/next.js/blob/7a1c9eb17eeea4e330e517455cb4c7a93cb3db2c/packages/next/lib/eslint/hasEslintConfiguration.ts#L45

Reply:
We also need to consider the case when eslint is defined in `package.json`

Reply:
Good point, updated in https://github.com/vercel/next.js/pull/39872/commits/775b6477ccadc2f83575c813e578946fa932d7d2